src/main/java/org/threadly/litesockets/buffers/SimpleMergedByteBuffers.java

```diff
@@ -0,0 +1,230 @@
+package org.threadly.litesockets.buffers;
+
+import java.nio.BufferUnderflowException;
+import java.nio.ByteBuffer;
+
+import org.threadly.util.ArgumentVerifier;
+
+/**
+ *  This is a lower overhead Implementation of {@link MergedByteBuffers}.
+ *  It is not appendable and can only process the buffers it is contructed with.
+ * 
+ */
+public class SimpleMergedByteBuffers extends AbstractMergedByteBuffers {
+  
+  private final ByteBuffer[] bba;
+  private int currentBuffer = 0;
+  protected long consumedSize = 0;
+  
+  public SimpleMergedByteBuffers(boolean readOnly, ByteBuffer ...bbs) {
+    super(readOnly);
+    for(ByteBuffer bb: bbs) {
+      if(bb == null) {
+        throw new IllegalArgumentException("Can not add null buffers!");
+      }
+    }
+    if(bbs.length > 0) {
+      bba = bbs;
+    } else {
+      bba = new ByteBuffer[] {EMPTY_BYTEBUFFER};
```

Comment:
Does it make since to have a static instance of this that's empty?  Not sure if we are hitting this case naturally or not.